[Decision] Add document for Elixir bindings (#4819)
ElektraInitiative/libelektra: doc/decisions/0_drafts/elixir_bindings.md

That is, every function of the Elektra C API which one wishes to expose to the Elixir binding has to be explicitly implemented as a NIF.
The big disadvantage is, of course, that this process is error-prone.
Furthermore, any change in the C API has to be manually carried over to the NIF.

### Alternative B

Use the existing code generation tool [Nifty](http://parapluu.github.io/nifty/).

Reasons for not choosing this approach are:

- Installation process required several patches of Nifty.
- Resource management is not that supplied by the Erlang NIF library but a custom one.

### Alternative C

Write a custom generation tool.

In principle one should be able to take a C header file and convert this to a NIF in an automatic fashion.

Solving this problem in general clearly would be a daunting task.
However, when restricting oneself to the use case for creating an interface for, say, Elektra's KDB this might actually have been feasible.
The main reson for not choosing this approach were time constraints.

## Decision

The chosen approach was to use "Alternative A", i.e., to implement the NIFs by hand.

## Rationale

The main reason for this choice were time constraints.

> Since I have assumed that the availability of the Elixir bindings is not very important, it is acceptable if future changes break these bindings.

Reviewer: See above, meaning "availability" needs to be clarified. (Here it reads like compatibility.)

## Implications

> - Any change in the C API breaks the Elixir binding which needs to be addressed manually.

Reviewer: ```suggestion
- Every incompatible change in the C API needs manual changes in the Elixir binding.
```
